packages/components/src/theme-type-badge/premium-badge/index.tsx

```diff
@@ -46,9 +46,25 @@ const PremiumBadge = ( {
 		  );
 
 	const divRef = useRef( null );
+	const labelRef = useRef( null );
 	const [ isPopoverVisible, setIsPopoverVisible ] = useState( false );
 	const [ isHovered, setIsHovered ] = useState( false );
 	const [ isPressed, setIsPressed ] = useState( false );
+	const [ displayLabelAsTooltip, setDisplayLabelAsTooltip ] = useState( false );
+	// This is used to prevent the label from being rendered in compact mode before the first render
+	// so that the label can be measured in its uncompacted state.
+	const [ mayRenderAsCompact, setMayRenderAsCompact ] = useState( false );
+
+	labelText = labelText || __( 'Premium' );
```

Comment:
This can override the given prop – not sure if it can cause some side effects on parent components that pass a value by reference which is used somewhere else.

Maybe we can do something like this instead to be on the safer side?

```diff
diff --git a/packages/components/src/theme-type-badge/premium-badge/index.tsx b/packages/components/src/theme-type-badge/premium-badge/index.tsx
index 7a37879c01..755c137342 100644
--- a/packages/components/src/theme-type-badge/premium-badge/index.tsx
+++ b/packages/components/src/theme-type-badge/premium-badge/index.tsx
@@ -22,7 +22,7 @@ interface BadgeProps {
 
 const PremiumBadge = ( {
        className,
-       labelText,
+       labelText: customLabelText,
        tooltipContent,
        tooltipClassName,
        tooltipPosition = 'bottom left',
@@ -55,7 +55,7 @@ const PremiumBadge = ( {
        // so that the label can be measured in its uncompacted state.
        const [ mayRenderAsCompact, setMayRenderAsCompact ] = useState( false );
 
-       labelText = labelText || __( 'Premium' );
+       const labelText = customLabelText ?? __( 'Premium' );
 
        // Display the label as a tooltip if the tooltip is being hidden and the label is too long.
        useLayoutEffect( () => {
~
```